timesketch/lib/datastores/elastic.py

```diff
@@ -452,6 +452,18 @@ def create_index(self, index_name=uuid4().hex, doc_type=u'generic_event'):
         doc_type = unicode(doc_type.decode(encoding=u'utf-8'))
         return index_name, doc_type
 
+    def delete_index(self, index_name):
+        """Delete Elasticsearch index.
+
+        Args:
+            index_name: Name of the index to delete.
+        """
+        if self.client.indices.exists(index_name):
+            try:
+                self.client.indices.delete(index=index_name)
+            except ConnectionError:
+                raise RuntimeError(u'Unable to connect to Timesketch backend.')
```

Comment:
Would it make sense to add specifics to the backend failing here?  With so much infrastructure, it might make it easier to troubleshoot.

Reply:
Yeah, this needs some more work overall, but I fixed this one. The log will show more details anyways but I included the specific connection error here.